### PR TITLE
Add documentation for configurations

### PR DESCRIPTION
**Pull request by gabrieltron**

### CODE REVIEW

**Comment by gabrieltron** [comment]
[[Docs] Update docs to include config](https://trello.com/c/vq7oygYJ/960-docs-update-docs-to-include-config)

File: docs/configurations_endpoints.md [hunk 1]
```diff
@@ -0,0 +1,554 @@
...
+  </TabItem>
+  <TabItem value="sh">
+
+  ```shell
+  TOKEN=YOUR_TOKEN
+  LEAF_USER_ID=00000000-0000-0000-0000-000000000000
+  
+  curl -X GET \
+      -H 'Authorization: Bearer ${YOUR_TOKEN}' \
+      'https://api.withleaf.io/services/config/api/configs/${LEAF_USER_ID}'
```
Comment (fabyan-alexander) [inline docs/configurations_endpoints.md:174]: Incorrect variable name and single quote won't replace the variable
```suggestion
  curl -X GET \
      -H "Authorization: Bearer ${TOKEN}" \
      "https://api.withleaf.io/services/config/api/configs/${LEAF_USER_ID}"
```

File: docs/configurations_endpoints.md [hunk 2]
```diff
@@ -0,0 +1,554 @@
...
+
+  </TabItem>
+  <TabItem value="sh">
+
+  ```shell
+  TOKEN=YOUR_TOKEN
+
+  curl -X GET \
+      -H 'Authorization: Bearer ${TOKEN}' \
+      'https://api.withleaf.io/service/configs/api/configs'
```
Comment (fabyan-alexander) [inline docs/configurations_endpoints.md:96]: Use double quote to allow the variables to be processed
```suggestion
  curl -X GET \
      -H "Authorization: Bearer ${TOKEN}" \
      "https://api.withleaf.io/service/configs/api/configs"
```

File: docs/configurations_endpoints.md [hunk 3]
```diff
@@ -0,0 +1,554 @@
...
+  <TabItem value="sh">
+
+  ```shell
+  TOKEN=YOUR_TOKEN
+  LEAF_USER_ID=00000000-0000-0000-0000-000000000000
+  
+  curl -X POST \
+      -H 'Authorization: Bearer ${TOKEN}' \
+      -d '{ "operationImageCreation": true, "operationsSummaryCreation": true, "fieldsAutoSync": true }' \
+      'https://api.withleaf.io/services/config/api/configs/${LEAF_USER_ID}'
```
Comment (fabyan-alexander) [inline docs/configurations_endpoints.md:276]: Use double quote to allow the variables to be processed
```suggestion
  curl -X POST \
      -H "Authorization: Bearer ${TOKEN}" \
      -d '{ "operationImageCreation": true, "operationsSummaryCreation": true, "fieldsAutoSync": true }' \
      "https://api.withleaf.io/services/config/api/configs/${LEAF_USER_ID}"
```

File: docs/configurations_endpoints.md [hunk 4]
```diff
@@ -0,0 +1,554 @@
...
+  </TabItem>
+  <TabItem value="sh">
+
+  ```shell
+  TOKEN = 'YOUR_TOKEN'
+
+  curl -X PUT \
+      -H 'Authorization: Bearer ${TOKEN}' \
+      -d '{ "operationImageCreation": true, "operationsSummaryCreation": true, "fieldsAutoSync": true }' \
+      'https://api.withleaf.io/services/config/api/configs'
```
Comment (fabyan-alexander) [inline docs/configurations_endpoints.md:374]: Use double quote to allow the variables to be processed
```suggestion
  curl -X PUT \
      -H "Authorization: Bearer ${TOKEN}" \
      -d '{ "operationImageCreation": true, "operationsSummaryCreation": true, "fieldsAutoSync": true }' \
      'https://api.withleaf.io/services/config/api/configs'
```

File: docs/configurations_endpoints.md [hunk 5]
```diff
@@ -0,0 +1,554 @@
...
+  <TabItem value="sh">
+
+  ```shell
+  TOKEN=YOUR_TOKEN
+  LEAF_USER_ID=00000000-0000-0000-0000-000000000000
+
+  curl -X PUT \
+      -H 'Authorization: Bearer ${TOKEN}' \
+      -d '{ "operationImageCreation": true, "operationsSummaryCreation": true, "fieldsAutoSync": true }' \
+      'https://api.withleaf.io/services/config/api/configs/${LEAF_USER_ID}'
```
Comment (fabyan-alexander) [inline docs/configurations_endpoints.md:476]: Use double quote to allow the variables to be processed
```suggestion
  curl -X PUT \
      -H "Authorization: Bearer ${TOKEN}" \
      -d '{ "operationImageCreation": true, "operationsSummaryCreation": true, "fieldsAutoSync": true }' \
      "https://api.withleaf.io/services/config/api/configs/${LEAF_USER_ID}"
```

File: docs/configurations_endpoints.md [hunk 6]
```diff
@@ -0,0 +1,554 @@
...
+  </TabItem>
+  <TabItem value="sh">
+
+  ```shell
+  TOKEN=YOUR_TOKEN
+  LEAF_USER_ID=00000000-0000-0000-0000-000000000000
+
+  curl -X DELETE \
+      -H 'Authorization: Bearer ${TOKEN}' \
+      'https://api.withleaf.io/services/config/api/configs/${LEAF_USER_ID}'
```
Comment (fabyan-alexander) [inline docs/configurations_endpoints.md:550]: Use double quote to allow the variables to be processed
```suggestion
  curl -X DELETE \
      -H "Authorization: Bearer ${TOKEN}" \
      "https://api.withleaf.io/services/config/api/configs/${LEAF_USER_ID}"
```

File: docs/configurations_endpoints.md [hunk 7]
```diff
@@ -0,0 +1,554 @@
...
+
+  </TabItem>
+</Tabs>
+
+### Create Leaf User's Configuration
+
+&nbsp<span class="badge badge--warning">POST</span> `/configs/{leafUserId}`
+
+Creates the Configuration for the Leaf User `leafUserId`. A resquest body must be provided
+containing the configurations to be set. All entries are optional, if missing the configuration will be inherited from the Api Owner's Configuration.
```
Comment (fabyan-alexander) [inline docs/configurations_endpoints.md:198]:
```suggestion
Creates the Configuration for the Leaf User `leafUserId`. A request body must be provided
containing the configurations to be set. All entries are optional, any missing configuration will be inherited from the Api Owner's Configuration.
```

File: docs/configurations_endpoints.md [hunk 8]
```diff
@@ -0,0 +1,554 @@
...
+  </TabItem>
+</Tabs>
+
+
+### Update Leaf User's Configuration
+
+&nbsp<span class="badge badge--success">PUT</span> `/configs/{leafUserId}`
+
+Updates the Configuration for the Leaf User `leafUserId`. A resquest body must be provided
+containing the configurations to be set. All entries are optional, if missing the configuration will be inherited from the Api Owner's Configuration.
```
Comment (fabyan-alexander) [inline docs/configurations_endpoints.md:399]:
```suggestion
Updates the Configuration for the Leaf User `leafUserId`. A request body must be provided
containing the configurations to be set. All entries are optional, any missing configuration will be inherited from the Api Owner's Configuration.
```

File: docs/configurations_endpoints.md [hunk 9]
```diff
@@ -0,0 +1,554 @@
...
+
+  </TabItem>
+</Tabs>
+
+### Update Api Owner's Configuration
+
+&nbsp<span class="badge badge--success">PUT</span> `/configs`
+
+Updates the Configuration for the Api Owner. A resquest body must be provided
+containing the configurations to be set. All entries are required.
```
Comment (fabyan-alexander) [inline docs/configurations_endpoints.md:300]:
```suggestion
Updates the Configuration for the Api Owner. A request body must be provided
containing the configurations to be set. All entries are required.
```

File: docs/configurations_endpoints.md [hunk 10]
```diff
@@ -0,0 +1,554 @@
...
+    "operationsSummaryCreation": true,
+    "operationImageCreation": true,
+    "fieldsAutoSync": true
+  }
+  ```
+
+  </TabItem>
+</Tabs>
+
+### Delete Leaf User's Configuraiton
```
Comment (fabyan-alexander) [inline docs/configurations_endpoints.md:495]:
```suggestion
### Delete Leaf User's Configuration
```

**Review by fabyan-alexander — CHANGES_REQUESTED** [review]
Nice update! Just a few typos and some small suggestions

**Review by fabyan-alexander — APPROVED** [review]
Approved

**Comment by fabyan-alexander** [comment]
[[Config] Fix typo in property name](https://trello.com/c/GVXu8Zo6/969-config-fix-typo-in-property-name)

**Review by CarlosSam — APPROVED** [review]

File: docs/configurations_endpoints.md [hunk 11]
```diff
@@ -0,0 +1,758 @@
...
+This service has the following endpoints available:
+
+Description | Endpoints
+--- | ---
+[Get Api Owner's Configuration][2] | <span class="badge badge--success">GET</span> `/configs`
+[Get Leaf User's Configuration][3] | <span class="badge badge--success">GET</span> `/configs/{leafUserId}`
+[Create Leaf User's Configuration][4] | <span class="badge badge--warning">POST</span> `/configs/{leafUserId}`
+[Override Api Owner's Configuration][5] | <span class="badge badge--success">PUT</span> `/configs`
+[Override Leaf User's Configuration][6] | <span class="badge badge--success">PUT</span> `/configs/{leafUserId}`
+[Update Leaf User's Configuraiton][7] | <span class="badge badge--warning">PATCH</span> `/configs/{leafUserId}`
```
Comment (fabyan-alexander) [inline docs/configurations_endpoints.md:39]:
```suggestion
[Update Api Owner's Configuraiton][7] | <span class="badge badge--warning">PATCH</span> `/configs/{leafUserId}`
```

**Review by fabyan-alexander — CHANGES_REQUESTED** [review]
Just a small fix

**Review by fabyan-alexander — APPROVED** [review]
Approved!

**Comment by gabrieltron** [comment]
[[Config] Update Postman and docs](https://trello.com/c/TwLHdTJZ/1153-config-update-postman-and-docs)

File: docs/configurations_endpoints.md [hunk 12]
```diff
@@ -0,0 +1,605 @@
...
+  <TabItem value="sh">
+
+  ```shell
+  TOKEN=YOUR_TOKEN
+  LEAF_USER_ID=00000000-0000-0000-0000-000000000000
+
+  curl -X PATCH \
+      -H 'Authorization: Bearer ${TOKEN}' \
+      -d '{ "operationsImageCreation": true, "fieldsAutoSync": true }' \
+      'https://api.withleaf.io/services/config/api/configs/${LEAF_USER_ID}'
```
Comment (fabyan-alexander) [inline docs/configurations_endpoints.md:514]: Variables inside a single quote won't be replaced 😉 
```suggestion
      "https://api.withleaf.io/services/config/api/configs/${LEAF_USER_ID}"
```

File: docs/configurations_endpoints.md [hunk 13]
```diff
@@ -0,0 +1,605 @@
...
+
+  </TabItem>
+  <TabItem value="sh">
+
+  ```shell
+  TOKEN=YOUR_TOKEN
+  LEAF_USER_ID=00000000-0000-0000-0000-000000000000
+
+  curl -X PATCH \
+      -H 'Authorization: Bearer ${TOKEN}' \
```
Comment (fabyan-alexander) [inline docs/configurations_endpoints.md:512]: Variables inside a single quote won't be replaced 😉 
```suggestion
      -H "Authorization: Bearer ${TOKEN}" \
```

File: docs/configurations_endpoints.md [hunk 14]
```diff
@@ -0,0 +1,605 @@
...
+  ```
+
+  </TabItem>
+  <TabItem value="sh">
+
+  ```shell
+  TOKEN = 'YOUR_TOKEN'
+
+  curl -X PATCH \
+      -H 'Authorization: Bearer ${TOKEN}' \
```
Comment (fabyan-alexander) [inline docs/configurations_endpoints.md:402]: Variables inside a single quote won't be replaced 😉 
```suggestion
      -H 'Authorization: Bearer ${TOKEN}' \
```

**Review by fabyan-alexander — CHANGES_REQUESTED** [review]
Nice update! Just a few fixed 😄

File: docs/configurations_endpoints.md (hunk 12 above)
Comment (gabrieltron) [reply docs/configurations_endpoints.md]: I was replicating a very nasty behavior that shouldn't be common in the docs, but it is, and that's not actually running those code snippets. I ran each of the snippets individually now, they all run successfully after some changes. We have more wrong snippets across the docs, this should be reviewed. bc35a888d7494b55fad60f05dc5c81adcc62c716

File: docs/configurations_endpoints.md (hunk 13 above)
Comment (gabrieltron) [reply docs/configurations_endpoints.md]: Fixed on bc35a888d7494b55fad60f05dc5c81adcc62c716

File: docs/configurations_endpoints.md (hunk 14 above)
Comment (gabrieltron) [reply docs/configurations_endpoints.md]: Fixed on bc35a888d7494b55fad60f05dc5c81adcc62c716

**Review by fabyan-alexander — APPROVED** [review]

File: docs/configurations_endpoints.md (hunk 12 above)
Comment (fabyan-alexander) [reply docs/configurations_endpoints.md]: Ouch, nice catch!

**Pull request merged**